components/fonts/platform/freetype/ohos/json.rs

```diff
@@ -0,0 +1,814 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use std::collections::{HashMap, HashSet};
+use std::convert::TryFrom;
+use std::fs;
+use std::path::PathBuf;
+use std::string::String;
+use std::vec::Vec;
+
+use serde::{Deserialize, Serialize};
+use serde_json::Value;
+
+use super::font_list::enumerate_font_files;
+
+static FONT_CONFIG_PATH: &str = "/etc/fontconfig.json";
+
+// This file contains custom functions for json parsing to avoid
+// blind reliance on serde_json::from_*. Even if user make some mistake in
+// fontconfig we want to inform him about the problem, but rest of usefull information
+// should be preserved and used in servo engine.
```

Comment:
It feels like maybe this is a bridge too far. It should be fine to not implement everything manually below and just report and error if the file cannot be parsed. The amount of parsing code below feels wrong somehow.